### ISSUE TITLE
[BUG] LocalBackend: Evaluation Failed!

### ISSUE THREAD

**Issue by jinmingteo** [awslabs/syne-tune#274]
Hi, I am using LocalBackend to train a couple of huggingface models for a sample dataset (still WIP)..

However, I ran into the following errors:
```
INFO:syne_tune.optimizer.schedulers.hyperband:trial_id 1 starts (first milestone = 1)
INFO:root:running subprocess with command: /opt/conda/bin/python huggingface_on_excel.py --model_type google/electra-base-discriminator --learning_rate 8.018154654725304e-05 --weight_decay 1.3591419560772573e-07 --dataset_path /DATA/jin/  --CUDA_VISIBLE_DEVICES 2 --train_batch_size 8 --valid_batch_size 8 --epochs 1 --output_dir output/ --eval_steps 100 --st_checkpoint_dir /root/syne-tune/test-hugging/1/checkpoints
INFO:syne_tune.tuner:(trial 1) - scheduled config {'model_type': 'google/electra-base-discriminator', 'learning_rate': 8.018154654725304e-05, 'weight_decay': 1.3591419560772573e-07, 'dataset_path': '/DATA/jin/, 'CUDA_VISIBLE_DEVICES': '2', 'train_batch_size': 8, 'valid_batch_size': 8, 'epochs': 1, 'output_dir': 'output/', 'eval_steps': 100}
INFO:syne_tune.tuner:Trial trial_id 1 was stopped independently of the scheduler.
INFO:syne_tune.optimizer.schedulers.fifo:trial_id 1: Evaluation failed!
```

Some of the debugging methods I have tried:
1. setting debug_mode : True in tuner did not reflect the bug.
2. I am able to run the exact commands for the subprocess without running into any issue or bug.

Any advice will be appreciated. Thank you!

**Comment by geoalgo** [comment]
Hi, 
Thanks for reporting this. 

1. Regarding "setting `debug_mode: True`", it is not clear to me what you mean since there is no such option in the Tuner. 
2. Without more context, it will be hard to guess exactly what is going on. Would it be possible to share the syne-tune script that you are using (how you are constructing and configuring the tuner). If this is not possible, would it be possible to let us know on which machine you are running (number cpu/gpu) and how many workers you are using in the tuner? 

One thing I suspect that could be happening is that your training script takes too much memory which could explain why running the command works but not the tuner as it runs several configurations in parallel.

**Comment by jinmingteo** [comment]
Hi @geoalgo, thank you for the prompt reply.

1. My mistake, it was ```"debug_log": True,``` but I am guessing that only reports scheduler bugs (?).
Speaking of which, are there other prompts to see what exactly is causing the evaluation to fail?

2. I am not sure how to share all the scripts but this is a snippet of my Tuner construction.
```
    n_workers = len(args.CUDA_VISIBLE_DEVICES.split(','))
    tuner = Tuner(
        trial_backend=LocalBackend(entry_point='huggingface.py', rotate_gpus=False),
        scheduler=MOBSTER(
            config_space,
            metric=f'eval_{args.metrics}',
            resource_attr='epoch',
            max_t=100,
            mode=args.metrics_mode,
            search_options={
                            "debug_log": True,
                            "num_init_random": n_workers + 2,
                            "gp_resource_kernel": "exp-decay-sum",  # GP surrogate model
                           },
        ),
        stop_criterion=StoppingCriterion(max_wallclock_time=args.timeout, 
                                         min_metric_value={args.metrics: args.min_metric_value}),
        n_workers=n_workers,
        max_failures=5,
    )
    tuner.run()
```

I am currently just experimenting on 1 GPU. Initially, I had the same suspicion as you but even after allocating 1 worker for 1 GPU and dropping the batch_size such that it was only using about half of the GPU memory, I am still running into evaluation failure.. Will continue to observe..
